src/main/java/com/cloudbees/plugins/credentials/impl/UsernamePasswordCredentialsImpl.java

```diff
@@ -72,6 +77,13 @@ public UsernamePasswordCredentialsImpl(@CheckForNull CredentialsScope scope,
         this.password = Secret.fromString(password);
     }
 
+    private Object readResolve() {
+        if (usernameSecret == null) {
+            usernameSecret = true;
```

Comment:
Default true for deserialized credentials, either from settings predating the upgrade, or REST uploads as XML.